src/components/ChannelSearch/ChannelSearch.tsx

```diff
@@ -109,14 +117,19 @@ const UnMemoizedChannelSearch = <
     return () => document.removeEventListener('click', clickListener);
   }, [resultsOpen]);
 
-  const selectResult = async (user: UserResponse<Us>) => {
+  const selectResult = async (result: ChannelOrUserType) => {
     if (!client.userID) return;
 
-    // @ts-expect-error
-    const newChannel = client.channel(channelType, { members: [client.userID, user.id] });
-    await newChannel.watch();
+    if (isChannel(result)) {
+      // @ts-expect-error
```

Comment:
the type guard, if working properly, should get rid of the need for the expect error, no?

Reply:
I meant to come back to this.

```Argument of type 'Channel<DefaultAttachmentType, DefaultChannelType, LiteralStringForUnion, UnknownType, DefaultMessageType, UnknownType, DefaultUserType<...>>' is not assignable to parameter of type 'Channel<At, Ch, Co, Ev, Me, Re, Us>'.
```